frontend/src/ui/SearchableSelect.tsx

```diff
@@ -114,27 +119,28 @@ export const theme = (theme: Theme) => ({
 
 
 
-type SeriesSelectorProps = DerivedProps<SeriesOption> & {
-    onChange?: (series: SeriesOption | null) => void;
+type VideoListSelectorProps = DerivedProps<VideoListOption> & {
+    onChange?: (videoList: VideoListOption | null) => void;
     onBlur?: () => void;
-    defaultValue?: SeriesOption;
+    defaultValue?: VideoListOption;
     writableOnly?: boolean;
+    type: "playlist" | "series";
 };
 
-type SeriesOption = {
-    readonly id: string;
-    readonly opencastId: string;
-    readonly title: string;
+export type VideoListOption = {
+    readonly id?: string;
+    readonly opencastId?: string;
+    readonly title?: string;
```

Comment:
Making these fields optional is actually incorrect. This type error pointed to an oversight, which is exactly what we want from Typescript. That's why it's best to first try to assume Typescript is right and points to a real problem instead of just trying to please the compiler. The problem here is in `EditMode/Playlist.tsx` where the playlist fields could be `undefined`. That is because the playlist might not be readable for the current user. The proper way to fix it is to deal with that case. For example like this (sorry for screenshot, but easiest way to show the diff):

![image](https://github.com/elan-ev/tobira/assets/7419664/94302a94-7761-41ef-8b87-063238bc2a83)

I mean it's not super nice from UI perspective and it means that if u cant read the playlist, u cannot change block settings. But this happens rare enough that it's fine I'd say. But yeah, the point being: in your current code, an unreadable playlist would lead to scrambled data and likely weird errors when saving or so.